merge.go

```diff
@@ -12,6 +12,12 @@ type Merger struct {
 	depth  int
 }
 
+// MergeAll ...
+func MergeAll(orig interface{}, n interface{}) interface{} {
+	m := &Merger{}
+	return m.mergeObj(orig, n, "$")
```

Comment:
Should this get passed a current path rather than "$", so debugging doesn't imply that there's merging going on in a `$.properties.this` key (using the new test case yaml as an example).